### ISSUE TITLE
Text field accessibility issues

### ISSUE THREAD

**Issue by betrue-final-final** [OfficeDev/office-ui-fabric-react#1618]
Environment: IE11 and JAWS 18
Are you willing to submit a PR to fix? No. I need help
Requested priority: High

1. Placeholder text is not announced when focus is on the text field.
Expect: Placeholder text should be read after the name and type of control, as a hint.

2. Validation errors don't announce themselves when they occur
Expect: Errors should be known to blind users

**Comment by scriby** [comment]
As far as I know the first issue is not an issue. I've been using text fields and the placeholders have been reading properly. There's certainly nothing strange that fabric-react is doing; they're just setting the placeholder attribute.

On the 2nd one, the approach I've found recommended is to associate the error message with the text box using aria-describedby. I'm not sure if that's supported in Narrator yet though.

**Comment by betrue-final-final** [comment]
I'll try it again on the placeholders, but with JAWS 18 it didn't read for me.

On the second issue, note that it was also using JAWS 18. Are you using that?

________________________________

**Comment by scriby** [comment]
Yes I am using Jaws 18, but testing in Chrome instead of IE11, so that may explain some differences. (Note that I have a specific requirement to test JAWS + Chrome for my product.)

**Comment by betrue-final-final** [comment]
Sure, but we need to work in as well.

**Comment by betrue-final-final** [comment]
I realize that possibly the placeholder audio is different from the placeholder text displayed. The display says "Now I am a Placeholder", but I do hear "Please enter text here". Can we match the placeholder audio with the display to avoid future confusion?

**Comment by mdahamiwal** [comment]
@betrue-final-final we did a lot of accessibility fixes for TextField including the error announcement. The main issue still remains that we can't always support all ATs+Browser combinations. 

This works fine with NVDA + Firefox and that is an industry standard as well. Fixing something for IE11 might again break it for other screen reader. As of IE11, there are a lot of bugs on the IE team to catch up with latest aria guidelines. Infact, many of aria 1.1 guidelines are not yet supported in IE11.

IMO, fabric team should have an internal discuss to draw the support boundaries otherwise it creates a lot of repetitive backlog and regression. Thoughts?

**Comment by betrue-final-final** [comment]
JAWS and IE11 are the de facto standard period. The National Federation of the Blind recommends this to their members and it's used by most blind users. @Anne<mailto:annetay@microsoft.com> if you would like more information or guidance.

I understand it's difficult, but this is the world where are customers are.

________________________________

**Comment by scriby** [comment]
http://webaim.org/projects/screenreadersurvey6/ has some stats on users of screen readers in the wild. This is a couple years old, but it's probably not too far off.

A couple takeaways:

* JAWS has (or had) the largest share, but is declining
* The highest used combo was (JAWS + IE) at 16%, and the next is NVDA + FF at 11%.

**Comment by betrue-final-final** [comment]
Looking at this with Narrator and Edge, I'm finding that only the first control announces its label.
Can we check that out?

**Comment by petdun2519** [comment]
@betrue-final-final and @mdahamiwal can you please review https://github.com/OfficeDev/office-ui-fabric-react/pull/2958

We had a bug filed on us where narrator was not announcing the error message for text field. After I made this change I can see it is being read aloud properly, think it should work for nvda and firefox as well but haven't tested. Found this issue when looking for similar issues to mine
